File: Core/Entities/OrderAggregate/Order.cs
namespace Core.Entities.OrderAggregate;

public class Order : BaseEntity
{
    public Order()
    {

    }
    public Order(IReadOnlyList<OrderItem> orderItems, string buyerEmail, Address shipToAddress,
    DeliveryMethod? deliveryMethod, decimal subtotal)
    {
        OrderItems = orderItems;
        BuyerEmail = buyerEmail;
        ShipToAddress = shipToAddress;
        DeliveryMethod = deliveryMethod;
        Subtotal = subtotal;
        // this.PaymentIntentId = paymentIntentId;
    }
    public string BuyerEmail { get; set; }
    public DateTimeOffset OrderDate { get; set; } = DateTimeOffset.Now;
    public Address ShipToAddress { get; set; }
    public DeliveryMethod? DeliveryMethod { get; set; }
    public IReadOnlyList<OrderItem> OrderItems { get; set; }
    public decimal Subtotal { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    public string PaymentIntentId { get; set; }
    public decimal Total { get; set; }
}

File: client/src/app/shared/models/order.ts
import { Address } from './address';

export interface OrderToCreate {
  basketId: string;
  deliveryMethodId: number;
  shipToAddress: Address;
}

export interface OrderItem {
  productId: number;
  productName: string;
  pictureUrl: string;
  price: number;
  quantity: number;
}

export interface Order {
  id: number;
  buyerEmail: string;
  orderDate: Date;
  shipToAddress: Address;
  deliveryMethod: string;
  shippingPrice: number;
  orderItems: OrderItem[];
  subtotal: number;
  status: string;
  total: number;
}
